pyfolio/perf_attrib.py

```diff
@@ -0,0 +1,148 @@
+        Returns by factor, with date as index and factors as columns
+        - Example:
+                        momentum  reversal
+            2017-01-01  0.002779 -0.005453
+            2017-01-02  0.001096  0.010290
+
+    strategy_daily_returns : pd.DataFrame
+        Returns for each day in the date range.
+        - Example:
+                            AAPL       TLT       XOM
+            2017-01-01 -0.014092 -0.003938 -0.001508
+            2017-01-02  0.011921  0.003039 -0.001546
+
+    strategy_daily_holdings: pd.Series
+        Daily holdings for all days in the date range, indexed by date
+        and ticker
+        - Example:
+            dt          ticker
+            2017-01-01  AAPL      71
+                        TLT       93
+                        XOM       10
+            2017-01-02  AAPL      71
+                        TLT       16
+                        XOM       71
+
+    Returns
+    -------
+    exposures : pd.DataFrame
+        df with factors as columns, and datetimes as index
+        - Example:
+                                 momentum    reversal
+            dt         ticker
+            2017-01-01 AAPL   -113.096901   60.550926
+                       TLT      17.192383   83.284658
+                       XOM       9.931600   11.493529
+            2017-01-02 AAPL     -9.940674  -37.271562
+                       TLT     -17.071643    2.966952
+                       XOM    -127.686503   54.070000
+
+    perf_attribution : pd.DataFrame
+        df with factors, common returns, and specific returns as columns,
+        and datetimes as index
+        - Example:
+                        momentum  reversal  common_returns  specific_returns
+            dt
+            2017-01-01  0.249087  0.935925        1.185012          1.185012
+            2017-01-02 -0.003194 -0.400786       -0.403980         -0.403980
+    """
+
+    risk_exposures = factor_loadings.multiply(strategy_daily_holdings,
+                                              axis='rows')
+    risk_exposures_portfolio = risk_exposures.groupby(level='dt').sum()
+    perf_attrib_style = risk_exposures_portfolio.multiply(factor_returns)
+
+    common_pa = perf_attrib_style.sum(axis=1)
+    total_returns_daily = strategy_daily_returns.sum(axis=1)
+    specific_pa = total_returns_daily - common_pa
+
+    perf_attrib_common = perf_attrib_style.assign(common_returns=common_pa)
+    perf_attrib = perf_attrib_common.assign(specific_returns=specific_pa)
+
+    return strategy_daily_holdings, risk_exposures, perf_attrib
+
+
+def create_perf_attrib_stats(strategy_daily_holdings, risk_exposures,
+                             perf_attrib):
+    """
+    Takes perf attribution data over a period of time and computes annualized
+    alpha, multifactor sharpe, risk exposures.
+    """
+
+    summary = {}
+    specific_returns = perf_attrib['specific_returns']
+    common_returns = perf_attrib['common_returns']
+
+    summary['daily_mf_alpha'] = (
```

Comment:
renamed things in `summary`,

re specific returns, @twiecki I thought so but one of @georgh0021 's formulas said:
`daily_mf_alpha = (specific_pnl / aums[date_range]).mean()`